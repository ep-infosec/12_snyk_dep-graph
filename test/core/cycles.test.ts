import * as depGraphLib from '../../src';
import * as helpers from '../helpers';

describe('cycles', () => {
  let depGraph;
  let toor;
  let foo;
  let bar;
  let baz;

  beforeEach(() => {
    const cyclicDepGraphData = helpers.loadFixture('cyclic-dep-graph.json');
    cyclicDepGraphData.graph.nodes
      .find((x) => x.nodeId === 'foo@2|x')
      .deps.push({ nodeId: 'baz@4|x' });
    depGraph = depGraphLib.createFromJSON(cyclicDepGraphData);
    [toor, foo, bar, baz] = depGraph.getPkgs();
  });

  test('pkgPathsToRoot - should work with cycles', () => {
    depGraph.getPkgs().forEach((pkg) => {
      const pkgPathsToRoot = depGraph.pkgPathsToRoot(pkg);

      expect(pkgPathsToRoot).toMatchSnapshot(`${pkg.name}@${pkg.version}`);
    });
  });

  describe('countPathsToRoot - should work with cycles', () => {
    it('toor', () => expect(depGraph.countPathsToRoot(toor)).toBe(1));
    it('foo', () => expect(depGraph.countPathsToRoot(foo)).toBe(1));
    it('bar', () => expect(depGraph.countPathsToRoot(bar)).toBe(1));
    it('baz', () => expect(depGraph.countPathsToRoot(baz)).toBe(2));
  });
});
