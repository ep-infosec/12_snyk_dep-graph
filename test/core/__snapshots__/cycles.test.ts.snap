// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cycles pkgPathsToRoot - should work with cycles: bar@3 1`] = `
Array [
  Array [
    Object {
      "name": "bar",
      "version": "3",
    },
    Object {
      "name": "foo",
      "version": "2",
    },
    Object {
      "name": "toor",
      "version": "1.0.0",
    },
  ],
]
`;

exports[`cycles pkgPathsToRoot - should work with cycles: baz@4 1`] = `
Array [
  Array [
    Object {
      "name": "baz",
      "version": "4",
    },
    Object {
      "name": "foo",
      "version": "2",
    },
    Object {
      "name": "toor",
      "version": "1.0.0",
    },
  ],
  Array [
    Object {
      "name": "baz",
      "version": "4",
    },
    Object {
      "name": "bar",
      "version": "3",
    },
    Object {
      "name": "foo",
      "version": "2",
    },
    Object {
      "name": "toor",
      "version": "1.0.0",
    },
  ],
]
`;

exports[`cycles pkgPathsToRoot - should work with cycles: foo@2 1`] = `
Array [
  Array [
    Object {
      "name": "foo",
      "version": "2",
    },
    Object {
      "name": "toor",
      "version": "1.0.0",
    },
  ],
]
`;

exports[`cycles pkgPathsToRoot - should work with cycles: toor@1.0.0 1`] = `
Array [
  Array [
    Object {
      "name": "toor",
      "version": "1.0.0",
    },
  ],
]
`;
